gordon/record_checker.py

```diff
@@ -0,0 +1,132 @@
+        """Extract record data from a record dict
+
+        Args:
+            record (dict): DNS record as a dict
+
+        Returns:
+            The record data: name, rrdata, type and ttl
+        """
+        return record.values()
+
+    async def check_record(self, record, timeout=60):
+        """Measures the time for a DNS record to become available.
+
+        Query a provided DNS server multiple time until
+            the reply matches the information in the
+            record or until timeout is reached.
+
+        Args:
+            record (dict): DNS record as a dict
+            with record properties
+            timeout (int): time threshold to query the DNS server
+        """
+        start_time = time.time()
+
+        name, rr_data, r_type, ttl = self._extract_record_data(record)
+        r_type_code = async_dns.types.get_code(r_type)
+
+        resolvable_record = False
+        retries = 0
+        sleep_time = 5
+
+        while not resolvable_record and \
+                timeout > retries * sleep_time:
+
+            retries += 1
+            resolver_res = await self._resolver.query(name, r_type_code)
+            possible_ans = resolver_res.an
+
+            resolvable_record = \
+                await self._check_resolver_ans(possible_ans, name,
+                                               rr_data, ttl, r_type_code)
+
+            if not resolvable_record:
+                await asyncio.sleep(sleep_time)
+
+        if not resolvable_record:
+            logging.info(
+                f'Sending metric record-checker-failed: {record}')
+        else:
+            final_time = float(time.time() - start_time)
+            success_msg = f'This record: {record} took {final_time} to register'
+            logging.info(success_msg)
+
+    async def _check_resolver_ans(
+            self, dns_answer_list, record_name,
+            record_ip_list, record_ttl, record_type_code):
+        """Check if resolver answer is equal to record data.
+
+        Args:
+            dns_answer_list (list): DNS answer list contains record objects
+            record_name (str): record name
+            record_ip_list (list): list of ips for the record
+            record_ttl (int): record time ot live info
+            record_type_code (int): record type code
+
+        Returns:
+            boolean indicating if DNS ans data is equal to record data
+        """
+        type_filtered_list = [
+            ans for ans in dns_answer_list if ans.qtype == record_type_code
+        ]
+
+        # check to see that type_filtered_lst has
+        # the same number of records as record_ip_list
+        if len(type_filtered_list) != len(record_ip_list):
+            return False
+
+        # check each record data is equal to the given data
+        for rec in type_filtered_list:
+            conditions = [rec.name == record_name,
+                          rec.ttl == record_ttl,
+                          rec.data in record_ip_list]
```

Comment:
I still think it would be better to `and` these together since then they'll be lazily evaluated, rather than evaluating them all first and then calling `all` on the result. @matt0bi What do you think?  I know you had a comment on `all` earlier, and I might be over-optimising since I don't know the use-case for this, what type of records, etc.

Reply:
Nice catch @descent3me! I do, however, think that using `all()` here makes the condition easier to read (I'm also a fan of using `all()`). From my own looking through DNS data, I think in something like 99.9% of the cases, the number of records in the `record_ip_list` will be pretty small, maybe a handful or two.